crates/ra_analysis/src/completion/complete_keyword.rs

```diff
@@ -18,6 +18,27 @@ pub(super) fn complete_expr_keyword(acc: &mut Completions, ctx: &CompletionConte
     if !ctx.is_trivial_path {
         return;
     }
+
+    // complete keyword "crate" in use stmt
+    if let (Some(use_item), None) = (&ctx.use_item_syntax, &ctx.path_prefix) {
+        let mut complete_crate = false;
+        let use_tree = use_item.use_tree();
+        if let Some(use_tree) = use_tree {
+            let tree_str = use_tree.syntax().text().to_string();
+            if tree_str != "crate" && "crate".starts_with(&tree_str) {
```

Comment:
Oh, this is a subtlety I should have explained in the docs: completion engine should **not** filter by the substring which is already present, it should give all possible variants for the identifier at the caret. In other words, 

for 

```
fn f() {
     let foo = 92;
     let _ = bar<|>
}
```

`foo` **should** be present among the completion variants. Filtering by identifier prefix/fuzzy match should be done higher in the stack.